bindings/azure/blobstorage/blobstorage.go

```diff
@@ -201,26 +231,113 @@ func (a *AzureBlobStorage) delete(blobURL azblob.BlockBlobURL, req *bindings.Inv
+		options.Details.Metadata = boolVal
+	} else if err != nil {
+		return nil, fmt.Errorf("error parsing metadata: %w", err)
+	}
+
+	if boolVal, err := req.GetMetadataAsBool(includeSnapshots); boolVal {
+		options.Details.Snapshots = boolVal
+	} else if err != nil {
+		return nil, fmt.Errorf("error parsing metadata: %w", err)
+	}
+
+	if boolVal, err := req.GetMetadataAsBool(includeUncommittedBlobs); boolVal {
+		options.Details.UncommittedBlobs = boolVal
+	} else if err != nil {
+		return nil, fmt.Errorf("error parsing metadata: %w", err)
+	}
+
+	if boolVal, err := req.GetMetadataAsBool(includeDeleted); boolVal {
+		options.Details.Deleted = boolVal
+	} else if err != nil {
+		return nil, fmt.Errorf("error parsing metadata: %w", err)
+	}
+
+	if intVal, err := req.GetMetadataAsInt64(maxResults, 32); intVal != 0 {
+		options.MaxResults = int32(intVal)
+	} else if err != nil {
+		return nil, fmt.Errorf("error parsing metadata: %w", err)
+	}
+
+	if val, ok := req.Metadata[prefix]; ok && val != "" {
+		options.Prefix = val
+	}
+
+	var initialMarker azblob.Marker
+	if val, ok := req.Metadata[marker]; ok && val != "" {
+		initialMarker = azblob.Marker{Val: &val}
 	} else {
-		name = uuid.New().String()
+		initialMarker = azblob.Marker{}
 	}
 
-	blobURL := a.containerURL.NewBlockBlobURL(name)
+	var blobs []azblob.BlobItem
+	metadata := map[string]string{}
+	ctx := context.Background()
+	for currentMaker := initialMarker; currentMaker.NotDone(); {
+		listBlob, err := a.containerURL.ListBlobsFlatSegment(ctx, currentMaker, options)
+		if err != nil {
+			return nil, fmt.Errorf("error listing blobs: %w", err)
+		}
+
+		blobs = append(blobs, listBlob.Segment.BlobItems...)
+
+		numBlobs := len(blobs)
+		currentMaker = listBlob.NextMarker
+		metadata[marker] = *currentMaker.Val
+		metadata["number"] = strconv.FormatInt(int64(numBlobs), 10)
+
+		if numBlobs == int(options.MaxResults) {
+			break
+		}
+	}
+
+	jsonResponse, err := json.Marshal(blobs)
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal blobs to json: %w", err)
+	}
+
+	return &bindings.InvokeResponse{
+		Data:     jsonResponse,
+		Metadata: metadata,
+	}, nil
+}
+
+func (a *AzureBlobStorage) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
 	switch req.Operation {
 	case bindings.CreateOperation:
-		return a.create(blobURL, req)
+		return a.create(a.getBlobURL(req.Metadata), req)
 	case bindings.GetOperation:
-		return a.get(blobURL, req)
+		return a.get(a.getBlobURL(req.Metadata), req)
 	case bindings.DeleteOperation:
-		return a.delete(blobURL, req)
+		return a.delete(a.getBlobURL(req.Metadata), req)
 	case bindings.ListOperation:
-		fallthrough
+		return a.list(req)
 	default:
 		return nil, fmt.Errorf("unsupported operation %s", req.Operation)
 	}
 }
+
+func (a *AzureBlobStorage) getBlobURL(metadata map[string]string) azblob.BlockBlobURL {
+	name := ""
+	if val, ok := metadata[blobName]; ok && val != "" {
+		name = val
+		delete(metadata, blobName)
```

Comment:
It is odd that this method modifies the map. Any reason?

Reply:
I have moved that part to the create function because this is the only method which forwards the metadata to Azure to store it with the blob